src/containers/share-modal/utils.ts

```diff
@@ -0,0 +1,61 @@
+import { ShareToTwitter } from 'common/models/Analytics'
+import { ShareModalContent } from 'common/store/ui/share-modal/types'
+import {
+  fullAlbumPage,
+  fullPlaylistPage,
+  fullProfilePage,
+  fullTrackPage
+} from 'utils/route'
+
+import { messages } from './messages'
+
+type ShareToTwitterEvent = Omit<ShareToTwitter, 'eventName' | 'source'>
+
+export const getTwitterShareText = (content: ShareModalContent) => {
+  let twitterText = ''
+  let link = ''
+  let analyticsEvent: ShareToTwitterEvent
+  switch (content.type) {
+    case 'track': {
```

Comment:
@milindNshah regarding using an enum here, it seems like we do both union types and enums, and i personally prefer union types on most cases cause its more lightweight. how about we have this as an item for a dev huddle, would love to have a discussion with everyone to figure out some guidelines?

Reply:
Sure sounds good, didn't see this before I left my comments but feel free to ignore for now